randovania/generator/filler/retcon.py

```diff
@@ -278,6 +278,18 @@ def debug_print_weighted_locations(all_locations_weighted: WeightedLocations):
         ))
 
 
+def should_be_starting_pickup(player: PlayerState, locations: WeightedLocations) -> bool:
+    cur_starting_pickups = player.num_starting_pickups_placed
+    minimum_starting_pickups = player.configuration.minimum_random_starting_pickups
+    maximum_starting_pickups = player.configuration.maximum_random_starting_pickups
+
+    result = cur_starting_pickups < minimum_starting_pickups or not locations
+    if player.count_self_locations(locations) == 0 and cur_starting_pickups < maximum_starting_pickups:
```

Comment:
would appreciate a comment along the lines of `prioritize placing starting pickups over off-world pickups` to clarify the intent here for anyone reading later